File: Python Course/Software Development Module - in Python/Week10/BinApp.py
"""
BinApp.py
Billy Cussen
03/10/2020
"""

#Objects
import gc
import Bin 
from Bin import myBin

#Variables
repeat = "y"
inputCheck = True

while repeat == "y":

    #Input - Bin Color?
    while inputCheck:
        try:
            binColor = str(input("Please enter your Bin Color (\"Green\", \"Black\" or \"Brown\"): "))
            binColor = binColor.lower()
        except:
            print("You need to enter the bins color!")
        else:
            if(binColor != "green" and binColor != "black" and binColor != "brown"):
                print("You need to enter \"Green\", \"Black\" or \"Brown\"!")
            else:
                inputCheck = False

    #Input - Apartment Number?
    while inputCheck == False:
        try:
            apartmentNumber = int(input("Please enter your Apartment Number: "))
        except:
            print("You need to enter a number for your apartment!")
        else:
            if(apartmentNumber <= 0 or apartmentNumber > 255):
                print("You need to enter an Integer between 1 and 255!")
            else:
                inputCheck = True

    #Process & Output
    myBinApp = myBin(binColor, apartmentNumber)
    myBinApp.computeCollectionDay()
    myBinApp = None
    gc.collect()

    #Input - Repeat?
    repeat = ""
    while repeat != "y" and repeat != "n":
        repeat = str(input("Please enter \"Y\" to continue or \"N\" to quit: "))
        repeat = repeat.lower()
        if repeat != "y" and repeat !="n":
            print("You need to enter \"y\" or \"n\"")
    
print("Thanks for using my App!")

File: Java/Software Development Module - Java/Week10/BinApp.java
/*
BinApp.java
Billy Cussen
20/07/2020
*/

import java.util.Scanner;
import java.util.InputMismatchException;

public class BinApp {

    public static void main(String[]args){

        //Variables
        String binColor = new String();
        int apartmentNumber = 0;
        String result;
        String restart;
        boolean inputCheckColor;
        boolean inputCheckApartmentNumber;

        //Objects
        Scanner keyboard;
        Bin myBin;

        //Input
        do{
            restart = new String();
            keyboard = new Scanner(System.in);

            do{
                inputCheckColor = true;
                keyboard = new Scanner(System.in);

                try{
                    System.out.print("Please enter your Bin Color(Green, Black or Brown): ");
                    binColor = keyboard.nextLine();

                    if(!(binColor.equalsIgnoreCase("Green") || binColor.equalsIgnoreCase("Black") || binColor.equalsIgnoreCase("Brown"))){
                        throw new InputMismatchException();
                    }

                }catch(InputMismatchException e){
                    System.out.println("Please enter Either Green, Black or Brown");
                    inputCheckColor = false;
                } finally{
                    keyboard = null;
                }
                
            }while(!inputCheckColor);

            do{
                inputCheckApartmentNumber = true;
                keyboard = new Scanner(System.in);

                try{
                    System.out.print("Please enter your Apartment Number: ");
                    apartmentNumber = keyboard.nextInt();

                    if(apartmentNumber <= 0){
                        throw new InputMismatchException();
                    }

                }catch(InputMismatchException e){
                    System.out.println("Please enter a valid Apartment Number (e.g. 1, 12)");
                    inputCheckApartmentNumber = false;
                } finally{
                    keyboard = null;
                }
                
            }while(!inputCheckApartmentNumber);

            //Process
            myBin = new Bin(binColor,apartmentNumber);
            result = myBin.computeAndGetDay();
            myBin = null;
            //Output
            System.out.println(result);

            do{
                
                keyboard = new Scanner(System.in);

                try{
                    System.out.print("Would you like to Retry (Y or N)?: ");
                    restart = keyboard.nextLine();

                    if(!(restart.equalsIgnoreCase("Y") || restart.equalsIgnoreCase("N"))){
                        throw new InputMismatchException();
                    }

                }catch(InputMismatchException e){
                    System.out.println("Please enter either Y or N");
                } finally{
                    keyboard = null;
                }

            }while(!(restart.equalsIgnoreCase("Y") || restart.equalsIgnoreCase("N")));

        }while(restart.equalsIgnoreCase("Y"));

    }

}